api/src/main/java/io/opentelemetry/trace/DefaultTracer.java

```diff
@@ -67,43 +67,32 @@ static NoopSpanBuilder create(String spanName) {
     }
 
     private boolean isRootSpan;
-    @Nullable private SpanContext spanContext;
+    @Nullable private Context parent;
 
     @Override
     public Span startSpan() {
-      if (spanContext == null && !isRootSpan) {
-        spanContext = TracingContextUtils.getCurrentSpan().getContext();
+      if (parent == null) {
+        parent = Context.current();
+      }
+      if (isRootSpan) {
+        parent = TracingContextUtils.withSpan(DefaultSpan.getInvalid(), parent);
       }
 
-      return spanContext != null && !SpanContext.getInvalid().equals(spanContext)
-          ? new DefaultSpan(spanContext)
-          : DefaultSpan.getInvalid();
-    }
-
-    @Override
-    public NoopSpanBuilder setParent(Span parent) {
-      Utils.checkNotNull(parent, "parent");
-      spanContext = parent.getContext();
-      return this;
-    }
-
-    @Override
-    public NoopSpanBuilder setParent(SpanContext remoteParent) {
-      Utils.checkNotNull(remoteParent, "remoteParent");
-      spanContext = remoteParent;
-      return this;
+      return new DefaultSpan(TracingContextUtils.getSpan(parent).getContext());
```

Comment:
For example, for a root span this looks almost like this series of expensive no-ops 

`TracingContextUtils.getSpan(TracingContextUtils.withSpan(DefaultSpan.getInvalid(), Context.current())).getContext()`

vs

`DefaultSpan.getInvalid()`

Reply:
Fixed in 0eb98299.